xbmc/pictures/metadata/ImageMetadataParser.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "ImageMetadata.h"
+
+#include <memory>
+#include <string>
+
+#include <exiv2/exiv2.hpp>
+
+class CImageMetadataParser
+{
+public:
+  CImageMetadataParser();
+  ~CImageMetadataParser() = default;
+
+  std::unique_ptr<ImageMetadata> ExtractMetadata(const std::string& picFileName);
```

Comment:
How about making this method static? There seems no point in instantiating a `CImageMetadataParser`, right?

<details>
<summary>Proposed change</summary>

```diff
diff --git a/xbmc/pictures/PictureInfoTag.cpp b/xbmc/pictures/PictureInfoTag.cpp
index 02857a2ddd..7c63f5db72 100644
--- a/xbmc/pictures/PictureInfoTag.cpp
+++ b/xbmc/pictures/PictureInfoTag.cpp
@@ -62,8 +62,7 @@ bool CPictureInfoTag::Load(const std::string &path)
   // Load by Kodi's included own way
   if (!m_isLoaded)
   {
-    CImageMetadataParser metadataParser;
-    std::unique_ptr<ImageMetadata> metadata = metadataParser.ExtractMetadata(path);
+    std::unique_ptr<ImageMetadata> metadata = CImageMetadataParser::ExtractMetadata(path);
     if (metadata)
     {
       m_imageMetadata = *(metadata.release());
diff --git a/xbmc/pictures/metadata/ImageMetadataParser.cpp b/xbmc/pictures/metadata/ImageMetadataParser.cpp
index e991b642d5..a2c14c49d4 100644
--- a/xbmc/pictures/metadata/ImageMetadataParser.cpp
+++ b/xbmc/pictures/metadata/ImageMetadataParser.cpp
@@ -72,13 +72,15 @@ std::unique_ptr<ImageMetadata> CImageMetadataParser::ExtractMetadata(const std::
   auto image = Exiv2::ImageFactory::open(outputBuffer.data(), readbytes);
   image->readMetadata();
 
+  CImageMetadataParser parser;
+
   // extract metadata
-  ExtractCommonMetadata(image);
+  parser.ExtractCommonMetadata(image);
 
-  ExtractExif(image->exifData());
-  ExtractIPTC(image->iptcData());
+  parser.ExtractExif(image->exifData());
+  parser.ExtractIPTC(image->iptcData());
 
-  return std::move(m_imageMetadata);
+  return std::move(parser.m_imageMetadata);
 }
 
 void CImageMetadataParser::ExtractCommonMetadata(std::unique_ptr<Exiv2::Image>& image)
diff --git a/xbmc/pictures/metadata/ImageMetadataParser.h b/xbmc/pictures/metadata/ImageMetadataParser.h
index b7fa623da1..8117c30a02 100644
--- a/xbmc/pictures/metadata/ImageMetadataParser.h
+++ b/xbmc/pictures/metadata/ImageMetadataParser.h
@@ -18,12 +18,13 @@
 class CImageMetadataParser
 {
 public:
-  CImageMetadataParser();
   ~CImageMetadataParser() = default;
 
-  std::unique_ptr<ImageMetadata> ExtractMetadata(const std::string& picFileName);
+  static std::unique_ptr<ImageMetadata> ExtractMetadata(const std::string& picFileName);
 
 private:
+  CImageMetadataParser();
+
   void ExtractCommonMetadata(std::unique_ptr<Exiv2::Image>& image);
   void ExtractExif(Exiv2::ExifData& exifData);
   void ExtractIPTC(Exiv2::IptcData& iptcData);
diff --git a/xbmc/pictures/metadata/test/TestMetadataExtraction.cpp b/xbmc/pictures/metadata/test/TestMetadataExtraction.cpp
index ff61ecc881..96852efa58 100644
--- a/xbmc/pictures/metadata/test/TestMetadataExtraction.cpp
+++ b/xbmc/pictures/metadata/test/TestMetadataExtraction.cpp
@@ -13,16 +13,6 @@
 
 #include <gtest/gtest.h>
 
-namespace
-{
-std::unique_ptr<ImageMetadata> ExtractMetadata(const std::string& path)
-{
-  CImageMetadataParser metadataParser;
-  return metadataParser.ExtractMetadata(path);
-}
-
-} // namespace
-
 class TestMetadataExtraction : public ::testing::Test
 {
 protected:
@@ -32,7 +22,7 @@ protected:
 TEST_F(TestMetadataExtraction, TestGPSImage)
 {
   auto path = XBMC_REF_FILE_PATH("xbmc/pictures/metadata/test/testdata/exifgps.jpg");
-  std::unique_ptr<ImageMetadata> metadata = ExtractMetadata(path);
+  std::unique_ptr<ImageMetadata> metadata = CImageMetadataParser::ExtractMetadata(path);
   // Extract GPS tags
   EXPECT_EQ(metadata->exifInfo.GpsLat, "49° 5' 47.469\" N");
   EXPECT_EQ(metadata->exifInfo.GpsLong, "9° 23' 14.630\" E");
@@ -63,7 +53,7 @@ TEST_F(TestMetadataExtraction, TestGPSImage)
 TEST_F(TestMetadataExtraction, TestIPTC)
 {
   auto path = XBMC_REF_FILE_PATH("xbmc/pictures/metadata/test/testdata/iptc.jpg");
-  std::unique_ptr<ImageMetadata> metadata = ExtractMetadata(path);
+  std::unique_ptr<ImageMetadata> metadata = CImageMetadataParser::ExtractMetadata(path);
   // Check the IPTC tags of this image
   EXPECT_TRUE(metadata->iptcInfo.Keywords.empty());
   EXPECT_EQ(metadata->iptcInfo.RecordVersion, "4");
```
</details>

Reply:
Yes makes total sense, thanks!